ui/index.d.ts

```diff
@@ -155,6 +155,20 @@ export interface IAzureParentTreeItem extends IAzureTreeItem, IChildProvider {
     pickTreeItem?(expectedContextValue: string): IAzureTreeItem | undefined;
 }
 
+/**
+ * Used to ensure a single invalid object does not prevent display of other valid objects
+ * Invalid objects will be shown with the error and the object's name. If the name cannot be determined for any invalid objects, a TreeItem will be added to the end with a generic label like "Some items cannot be displayed"
+ * @param sourceArray The collection of source objects before converting to TreeItems
+ * @param invalidContextValue The context value to use for invalid source objects
+ * @param createTreeItem A function that converts a source object to a TreeItem
```

Comment:
Note that returning undefined from createTreeItem will simply skip that item (not added to tree)